src/libNode/FinalBlockProcessing.cpp

```diff
@@ -725,7 +727,10 @@ bool Node::ProcessFinalBlockCore(const bytes& message, unsigned int offset,
     // Remove because shard nodes will be shuffled in next epoch.
     CleanMicroblockConsensusBuffer();
 
-    StoreFinalBlock(txBlock);
+    if (!StoreFinalBlock(txBlock)) {
+      LOG_GENERAL(WARNING, "StoreFinalBlock failed!");
```

Comment:
How about move the error message into `StoreFinalBlock` then we can handle the debug message in one place for easier maintenance. For caller here, it would be simpler to check the function result only.